src/link/core/types.ts

```diff
@@ -76,7 +76,10 @@ export interface Operation {
   variables: Record<string, any>;
   operationName: string;
   extensions: Record<string, any>;
-  setContext: (context: DefaultContext) => DefaultContext;
+  setContext: {
+    (context: DefaultContext): void;
+    (updateContext: (previousContext: DefaultContext) => DefaultContext): void;
+  };
```

Comment:
```suggestion
  setContext: {
    (context: Partial<DefaultContext>): void;
    (
      updateContext: (
        previousContext: DefaultContext
      ) => Partial<DefaultContext>
    ): void;
  };
```

While we're at it, these should be `Partial` :)